core/include/moveit/task_constructor/stages/deep_grasp_pose.h

```diff
@@ -0,0 +1,310 @@
+	ROS_INFO_NAMED(LOGNAME, "Generate grasp goal now active");
+	found_candidates_ = false;
+}
+
+template <class ActionSpec>
+void DeepGraspPose<ActionSpec>::feedbackCallback(const FeedbackConstPtr& feedback) {
+	// each candidate should have a cost
+	if (feedback->grasp_candidates.size() != feedback->costs.size()) {
+		ROS_ERROR_NAMED(LOGNAME, "Invalid input: each grasp candidate needs an associated cost");
+	} else {
+		ROS_INFO_NAMED(LOGNAME, "Grasp generated feedback received %lu candidates: ", feedback->grasp_candidates.size());
+
+		found_candidates_ = true;
+
+		grasp_candidates_.resize(feedback->grasp_candidates.size());
+		costs_.resize(feedback->costs.size());
+
+		grasp_candidates_ = feedback->grasp_candidates;
+		costs_ = feedback->costs;
+	}
+}
+
+template <class ActionSpec>
+void DeepGraspPose<ActionSpec>::doneCallback(const actionlib::SimpleClientGoalState& state,
+                                             const ResultConstPtr& result) {
+	if (state == actionlib::SimpleClientGoalState::SUCCEEDED) {
+		ROS_INFO_NAMED(LOGNAME, "Found grasp candidates (result): %s", result->grasp_state.c_str());
+	} else {
+		ROS_ERROR_NAMED(LOGNAME, "No grasp candidates found (state): %s",
+		                ActionBaseT::clientPtr_->getState().toString().c_str());
+	}
+}
+
+template <class ActionSpec>
+void DeepGraspPose<ActionSpec>::init(const core::RobotModelConstPtr& robot_model) {
+	InitStageException errors;
+	try {
+		GeneratePose::init(robot_model);
+	} catch (InitStageException& e) {
+		errors.append(e);
+	}
+
+	const auto& props = properties();
+
+	// check angle_delta
+	if (props.get<double>("angle_delta") == 0.) {
+		errors.push_back(*this, "angle_delta must be non-zero");
+	}
+
+	// check availability of object
+	props.get<std::string>("object");
+	// check availability of eef
+	const std::string& eef = props.get<std::string>("eef");
+	if (!robot_model->hasEndEffector(eef)) {
+		errors.push_back(*this, "unknown end effector: " + eef);
+	} else {
+		// check availability of eef pose
+		const moveit::core::JointModelGroup* jmg = robot_model->getEndEffector(eef);
+		const std::string& name = props.get<std::string>("pregrasp");
+		std::map<std::string, double> m;
+		if (!jmg->getVariableDefaultPositions(name, m)) {
+			errors.push_back(*this, "unknown end effector pose: " + name);
+		}
+	}
+
+	if (errors) {
+		throw errors;
+	}
+}
+
+template <class ActionSpec>
+void DeepGraspPose<ActionSpec>::compute() {
+	if (upstream_solutions_.empty()) {
+		return;
+	}
+	planning_scene::PlanningScenePtr scene = upstream_solutions_.pop()->end()->scene()->diff();
+
+	// set end effector pose
+	const auto& props = properties();
+	const std::string& eef = props.get<std::string>("eef");
+	const moveit::core::JointModelGroup* jmg = scene->getRobotModel()->getEndEffector(eef);
+
+	robot_state::RobotState& robot_state = scene->getCurrentStateNonConst();
+	robot_state.setToDefaultValues(jmg, props.get<std::string>("pregrasp"));
```

Comment:
Should these pose names be set as `constexpr`s at the top? Not saying it's necessarily better that way, but suggesting that it be considered.

Reply:
I am not sure about this one

Reply:
No worries--I'm sure one of the friendly local maintainers will let you know if they want it changed.